roles/pgbouncer/config/tasks/main.yml

```diff
@@ -19,33 +19,34 @@
   when: existing_pgcluster is not defined or not existing_pgcluster|bool
   tags: pgbouncer, pgbouncer_conf
 
-- name: Get users and password md5 from pg_shadow
-  run_once: true
-  become: true
-  become_user: postgres
-  command: >
-    {{ postgresql_bin_dir }}/psql -p {{ postgresql_port }} -U {{ patroni_superuser_username }} -d postgres -Atq
-    -c "SELECT concat('\"', usename, '\" \"', passwd, '\"') FROM pg_shadow where usename != '{{ patroni_replication_username }}'"
-  register: pg_shadow_result
-  changed_when: false
-  delegate_to: "{{ groups.master[0] }}"
-  when: pgbouncer_generate_userlist|bool
-  tags: pgbouncer, pgbouncer_generate_userlist
+- block:  # when pgbouncer_auth_user false
```

Comment:
I think there is no need for a block here because we need to add the task "`Get users and password md5 from pg_shadow for user {{ pgbouncer_auth_username }}`" so as not to store the password in plain text in the userlist.txt file

Reply:
Well, as I said this does not work if the password is hashed here. See this issue: https://github.com/pgbouncer/pgbouncer/issues/774

Reply:
Thanks for the link, I subscribed to this issue 

hmmm, It's a pity that this is really the case and auth_user only works for md5, but not for SCRAM.

Reply:
Yeah, this is a bummer. The question is how to proceed with this PR? At the end of the day nobody wants plaintext passwords in text files.

Reply:
the alternative is 'trust' in pg_hba.conf for the 'pgbouncer' user. We can afford it for local pgbouncer.
In this case, there is no need to configure auth_file if auth_user is used.

Reply:
Hmm...I'm afraid this doesn't work:

```bash
2023-07-08 20:06:22.855 UTC [11474] WARNING server login failed: FATAL password authentication failed for user "pgbouncer"
2023-07-08 20:06:34.501 UTC [11474] LOG stats: 0 xacts/s, 0 queries/s, in 0 B/s, out 0 B/s, xact 0 us, query 0 us, wait 0 us
2023-07-08 20:06:37.171 UTC [11474] WARNING C-0x561666844e00: testdb/(nouser)@unix(11476):6432 pooler error: query_wait_timeout
```
`pg_hba.conf` 

```
# TYPE      DATABASE                 USER                     ADDRESS                  METHOD
  local     all                      postgres                                          trust
  local     all                      pgbouncer                                         trust
```

Reply:
See pool config

Use 'host', no 'local'

Reply:
postgresql_pg_hba:
  - { type: "local", database: "all", user: "{{ patroni_superuser_username }}", address: "", method: "trust" }
  - { type: "local", database: "replication", user: "{{ patroni_superuser_username }}", address: "", method: "trust" }
  - { type: "local", database: "all", user: "all", address: "", method: "peer" }
   - { type: "host", database: "all", user: "all", address: "127.0.0.1/32", method: "{{ postgresql_password_encryption_algorithm }}" }
  - { type: "host", database: "all", user: "pgbouncer", address: "127.0.0.1/32", method: "trust" } # required for pgbouncer auth_user
  - { type: "host", database: "all", user: "all", address: "::1/128", method: "{{ postgresql_password_encryption_algorithm }}" }

Reply:
yeah, just realised that when I saw the posgresql logs.